pkg/asset/installconfig/gcp/validation.go

```diff
@@ -133,3 +134,34 @@ func validateMachineNetworksContainIP(fldPath *field.Path, networks []types.Mach
 	}
 	return field.ErrorList{field.Invalid(fldPath, subnetName, fmt.Sprintf("subnet CIDR range start %s is outside of the specified machine networks", ip))}
 }
+
+//ValidateEnabledServices gets all the enabled services for a project and validate if any of the required services are not enabled.
+func ValidateEnabledServices(ctx context.Context, client API, project string) error {
+	services := sets.NewString("compute.googleapis.com",
+		"cloudapis.googleapis.com",
+		"cloudresourcemanager.googleapis.com",
+		"dns.googleapis.com",
+		"iam.googleapis.com",
+		"iamcredentials.googleapis.com",
+		"servicemanagement.googleapis.com",
+		"serviceusage.googleapis.com",
+		"storage-api.googleapis.com",
+		"storage-component.googleapis.com")
+	projectServices, err := client.GetEnabledServices(ctx, project)
+
+	if err != nil {
+		if apiError, ok := err.(*googleapi.Error); ok {
+			if apiError.Code == http.StatusForbidden {
+				logrus.Warn("Permission denied. Unable to fetch enabled services for project.")
+				return nil
+			}
+		}
```

Comment:
```suggestion
                var gErr *googleapi.Error
		if errors.As(err, &gErr) {
			if gErr.Code == http.StatusForbidden {
				logrus.Warn("Permission denied. Unable to fetch enabled services for project.")
				return nil
			}
		}
```

using errors.As compared to using type coercion (<err>.(type)) provides the benefit of unwrap and checking.